src/DebugPrintMacros.h

```diff
@@ -27,18 +27,34 @@ inline struct _DEBUG_TIME_STAMP debugTimeStamp(void) {
 }
 #endif
 
+#if defined(DEBUG_ESP_PORT) && !defined(DEBUG_ESP_PORT_PRINTF)
+
+#ifdef __cplusplus
+#define DEBUG_ESP_PORT_PRINTF(format, ...)   DEBUG_ESP_PORT.printf((format), ##__VA_ARGS__)
+#define DEBUG_ESP_PORT_PRINTF_F(format, ...) DEBUG_ESP_PORT.printf_P(PSTR(format), ##__VA_ARGS__)
+#define DEBUG_ESP_PORT_FLUSH DEBUG_ESP_PORT.flush
+#else
+// Handle debug printing from .c without CPP Stream, Print, ... classes
+// Cannot handle flash strings in this setting
+#define DEBUG_ESP_PORT_PRINTF ets_uart_printf
+#define DEBUG_ESP_PORT_PRINTF_F ets_uart_printf
+#define DEBUG_ESP_PORT_FLUSH (void)0
+#endif
+
+#endif
+
 #if defined(DEBUG_ESP_PORT) && !defined(DEBUG_GENERIC)
   #define DEBUG_GENERIC( module, format, ... ) \
     do { \
       struct _DEBUG_TIME_STAMP st = debugTimeStamp(); \
-      DEBUG_ESP_PORT.printf( DEBUG_TIME_STAMP_FMT module " " format, st.whole, st.dec, ##__VA_ARGS__ ); \
+      DEBUG_ESP_PORT_PRINTF( (DEBUG_TIME_STAMP_FMT module " " format), st.whole, st.dec, ##__VA_ARGS__ ); \
```

Comment:
Why these changes relating to logging?

I'd prefer them to be filed in a separate PR to make reviewing easier.

Reply:
Some changes were needed to generalize the debug print macro a little more so I could get them to work from tcp_axtls.c.